forbidigo/forbidigo.go

```diff
@@ -164,24 +192,44 @@ func (v *visitor) Visit(node ast.Node) ast.Visitor {
 		}
 		return v
 	case *ast.SelectorExpr:
+		selectorExpr = node
 	case *ast.Ident:
 	default:
 		return v
 	}
+
+	// The text as it appears in the source is always used because issue
+	// use that. The other texts to match against are extracted when needed
+	// by a pattern.
+	srcText := v.textFor(node)
+	pkgText := ""
 	for _, p := range v.linter.patterns {
-		if p.pattern.MatchString(v.textFor(node)) && !v.permit(node) {
-			v.issues = append(v.issues, UsedIssue{
-				identifier: v.textFor(node),
-				pattern:    p.pattern.String(),
-				pos:        node.Pos(),
-				position:   v.fset.Position(node.Pos()),
-				customMsg:  p.msg,
-			})
+		switch {
+		case p.matchPackage:
+			if pkgText == "" {
+				pkgText = v.textWithPackage(selectorExpr)
+			}
+			v.match(node, pkgText, srcText, p)
+		default:
+			v.match(node, srcText, srcText, p)
 		}
 	}
 	return nil
 }
 
+func (v *visitor) match(node ast.Node, text, srcText string, p *pattern) {
```

Comment:
I'd still favor inlining this given that it's hard to write a docstring that describes what function it performs.  I think all that needs to happen is to set up the value of `text` if we inline it.

Reply:
As additional justification, this method has a side-effect (appending to v.issues) that also isn't really described by the method name.

Reply:
Okay.